weaviate/connect/connection.py

```diff
@@ -174,12 +176,9 @@ def periodic_refresh_token(refresh_time: int, _auth: Optional[_Auth]):
             name="TokenRefresh",
         )
         demon.start()
-        self._shutdown_background_event = Event()
 
-    def __del__(self):
```

Comment:
Maybe we should add the `__del__` as well and call the `self.close()` there, this is just in case things go wrong and the `Client ` does not call the `self._connection.close()`.
What do you think?

Reply:
hmm. in what circumstance would the client `__del__` not be called, but then the connection `__del__`?

Reply:
There should not be such a case.

I have noticed that we need to add a `__del__` for `WCS` class as well, which should close the connection.

Reply:
https://github.com/semi-technologies/weaviate-python-client/blob/main/weaviate/wcs/crud_wcs.py#L18